proxy/src/serverless/tls_listener.rs

```diff
@@ -0,0 +1,283 @@
+use std::{
+    pin::Pin,
+    task::{Context, Poll},
+    time::Duration,
+};
+
+use futures::{Future, Stream, StreamExt};
+use pin_project_lite::pin_project;
+use thiserror::Error;
+use tokio::{
+    io::{AsyncRead, AsyncWrite},
+    task::JoinSet,
+    time::timeout,
+};
+
+/// Default timeout for the TLS handshake.
+pub const DEFAULT_HANDSHAKE_TIMEOUT: Duration = Duration::from_secs(10);
```

Comment:
Why don't you use cli argements instead?

Reply:
This was copied in from the library. I decided not to touch this part for now and just keep it the same. it's a good follow up item. I was thinking though instead to make the timeouts more restrictive with higher load.